datatypes/src/raster/grid_bounds.rs

```diff
@@ -0,0 +1,307 @@
+use snafu::ensure;
```

Comment:
should add some tests for all these calculations in this file

Reply:
done